script/discourse

```diff
@@ -56,210 +61,201 @@ class DiscourseCLI < Thor
     end
   end
 
   desc "backup", "Backup a discourse forum"
+  option :s3_uploads, type: :boolean, default: false, desc: "Include s3 uploads in the backup"
   def backup(filename = nil)
     load_rails
-    require "backup_restore"
-    require "backup_restore/backuper"
 
     store = BackupRestore::BackupStore.create
 
     if filename
-      destination_directory = File.dirname(filename).sub(/^\.$/, '')
+      destination_directory = File.dirname(filename).sub(/^\.$/, "")
 
       if destination_directory.present? && store.remote?
         puts "Only local backup storage supports paths."
         exit(1)
       end
 
-      filename_without_extension = File.basename(filename).sub(/\.(sql\.)?(tar\.gz|t?gz)$/i, '')
+      filename_without_extension = File.basename(filename).sub(/\.(sql\.)?(tar\.gz|t?gz)$/i, "")
     end
 
-    puts "Starting backup..."
-    backuper = BackupRestore::Backuper.new(Discourse.system_user.id, filename: filename_without_extension)
-    backup_filename = backuper.run
-    exit(1) unless backuper.success
+    old_include_s3_uploads_in_backup = SiteSetting.include_s3_uploads_in_backups
+    SiteSetting.include_s3_uploads_in_backups = true if options[:s3_uploads]
 
-    puts "Backup done."
+    begin
+      puts "Starting backup..."
+      backuper =
+        BackupRestore::Backuper.new(Discourse.system_user.id, filename: filename_without_extension)
+      backup_filename = backuper.run
+      exit(1) unless backuper.success
+
+      puts "Backup done."
+
+      if store.remote?
+        location =
+          BackupLocationSiteSetting.values.find { |v| v[:value] == SiteSetting.backup_location }
+        location = I18n.t("admin_js.#{location[:name]}") if location
+        puts "Output file is stored on #{location} as #{backup_filename}", ""
+      else
+        backup = store.file(backup_filename, include_download_source: true)
 
-    if store.remote?
-      location = BackupLocationSiteSetting.values.find { |v| v[:value] == SiteSetting.backup_location }
-      location = I18n.t("admin_js.#{location[:name]}") if location
-      puts "Output file is stored on #{location} as #{backup_filename}", ""
-    else
-      backup = store.file(backup_filename, include_download_source: true)
+        if destination_directory.present?
+          puts "Moving backup file..."
+          backup_path = File.join(destination_directory, backup_filename)
+          FileUtils.mv(backup.source, backup_path)
+        else
+          backup_path = backup.source
+        end
 
-      if destination_directory.present?
-        puts "Moving backup file..."
-        backup_path = File.join(destination_directory, backup_filename)
-        FileUtils.mv(backup.source, backup_path)
-      else
-        backup_path = backup.source
+        puts "Output file is in: #{backup_path}", ""
       end
-
-      puts "Output file is in: #{backup_path}", ""
+    ensure
+      SiteSetting.include_s3_uploads_in_backups = old_include_s3_uploads_in_backup
```

Comment:
This is **part 3/3** of the changes introduced in this PR. Ensuring we restore the previous value of the site setting.